vumi/persist/tests/test_fake_redis.py

```diff
@@ -1,491 +1,750 @@
+
+    def __getattr__(self, name):
+        return lambda *args, **kw: self._perform_operation(name, *args, **kw)
+
+
+class TestFakeRedisVerify(FakeRedisTestMixin, VumiTestCase):
+
+    if 'VUMITEST_REDIS_DB' not in os.environ:
+        skip = ("This test requires a real Redis server. Set VUMITEST_REDIS_DB"
+                " to run it.")
+
+    def get_redis(self, **kwargs):
+        from vumi.persist.redis_manager import RedisManager
+        # Fake redis
+        fake_redis = FakeRedis(**kwargs)
+        self.add_cleanup(fake_redis.teardown)
+        # Real redis
+        config = {
+            'FAKE_REDIS': 'yes',
+            'key_prefix': 'redistest',
+        }
+        config.update(kwargs)
+        real_redis = RedisManager.from_config(config)
+        self.add_cleanup(self.cleanup_manager, real_redis)
+        real_redis._purge_all()
+        # Both redises
+        return RedisPairWrapper(self, fake_redis, real_redis)
+
+    def cleanup_manager(self, manager):
+        manager._purge_all()
+        manager._close()
+
+    def assert_redis_op(self, redis, expected, op, *args, **kw):
         self.assertEqual(expected, getattr(redis, op)(*args, **kw))
 
-    @inlineCallbacks
-    def test_charset_encoding_default(self):
-        # Redis client assumes utf-8
-        redis = self.get_redis()
-        yield redis.set('name', u'Zoë Destroyer of Ascii')
-        yield self.assert_redis_op(redis, 'Zo\xc3\xab Destroyer of Ascii',
-            'get', 'name')
+    def assert_redis_error(self, redis, op, *args, **kw):
+        self.assertRaises(Exception, getattr(redis, op), *args, **kw)
 
-    @inlineCallbacks
-    def test_charset_encoding_custom_replace(self):
-        redis = self.get_redis(charset='ascii', errors='replace')
-        yield redis.set('name', u'Zoë Destroyer of Ascii')
-        yield self.assert_redis_op(redis, 'Zo? Destroyer of Ascii',
-            'get', 'name')
+    def wait(self, redis, delay):
+        redis._fake_redis.clock.advance(delay)
+        d = Deferred()
+        reactor.callLater(delay, d.callback, None)
+        return d
+
+
+class TestFakeRedisVerifyAsync(FakeRedisTestMixin, VumiTestCase):
+
+    if 'VUMITEST_REDIS_DB' not in os.environ:
+        skip = ("This test requires a real Redis server. Set VUMITEST_REDIS_DB"
+                " to run it.")
 
     @inlineCallbacks
-    def test_charset_encoding_custom_ignore(self):
-        redis = self.get_redis(charset='ascii', errors='ignore')
-        yield redis.set('name', u'Zoë Destroyer of Ascii')
-        yield self.assert_redis_op(redis, 'Zo Destroyer of Ascii',
-            'get', 'name')
+    def get_redis(self, **kwargs):
+        from vumi.persist.txredis_manager import TxRedisManager
+        # Fake redis
+        fake_redis = FakeRedis(async=True, **kwargs)
+        self.add_cleanup(fake_redis.teardown)
+        # Real redis
+        config = {
+            'FAKE_REDIS': 'yes',
+            'key_prefix': 'redistest',
+        }
```

Comment:
If I'm understanding this correctly setting `FAKE_REDIS` to `'yes'` would cause us to create a fake redis instance, even though this config is for real redis. Any reason why we do that here?